tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,139 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Managing Virtual Networks with virsh
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+
+sub run {
+
+    #Prepare New Guest Network Interface Configuration
+    foreach my $guest (keys %xen::guests) {
+        assert_script_run("ssh root\@$guest 'cd /etc/sysconfig/network/; cp ifcfg-eth0 ifcfg-eth1'", 60);
+    }
+
+#NAT BASED NETWORK
+#Prepare Test
+my $vnet_nated = <<'EOF';
+<network>
+ <name>vnet_nated</name>
+ <bridge name='virbr1' stp='on' delay='0'/>
+ <forward mode='nat'/>
+ <ip address="192.168.122.1" netmask="255.255.255.0">
+  <dhcp>
+   <range start="192.168.122.2" end="192.168.122.254" />
+  </dhcp>
+ </ip>
+</network>
+EOF
+    foreach my $guest (keys %xen::guests) {
+        record_info "virsh-virtual_network", "NAT BASED NETWORK";
+	assert_script_run("echo '$vnet_nated' >> vnet_nated.xml");
+	assert_script_run("virsh net-create vnet_nated.xml");
+	assert_script_run("virsh net-info vnet_nated");
+	assert_script_run("virsh attach-interface $guest network vnet_nated --live");
+	#my $gi_vnet_nated = script_output("sleep 30;journalctl | grep dnsmasq-dhcp| tail -1| grep -Po "\d+\.\d+\.\d+\.\d+"");
+	assert_script_run("ssh root\@$guest 'ping -c5 -W1 -I eth1 10.160.0.207'",60);
+	save_screenshot;
+	assert_script_run("virsh detach-interface $guest network --current");
+	assert_script_run("virsh net-destroy vnet_nated");
+    }
+
+#BRIDGE - USING AN EXISTING BRIDGE ON VM HOST SERVER
+#Prepare Test
+my $vnet_host_bridge = <<'EOF';
+<network>
+        <name>vnet_host_bridge</name>
+        <forward mode="bridge"/>
+        <bridge name="br0"/>
+</network>
+EOF
+    foreach my $guest (keys %xen::guests) {
+	record_info "virsh-virtual_network", "BRIDGE - USING AN EXISTING BRIDGE ON VM HOST SERVER";
+        assert_script_run("echo '$vnet_host_bridge' >> vnet_host_bridge.xml");
+        assert_script_run("virsh net-create vnet_host_bridge.xml");
+        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --live");
+        my $mac_host_bridge = script_output("sleep 30;virsh domiflist $guest | grep vnet_host-bridge| grep -E -o '[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}'");
+        my $gi_host_bridge = script_output("arp | grep $mac_host_bridge| awk \'{print $1}\'");
+        assert_script_run("ssh root\@$gi_host_bridge 'ping -c5 -W1 -I eth1 10.160.0.207'",60);
```

Comment:
Ping should come out via the nic of this tested network. How do you know it is eth1? Better to replace fixed string with some nic finding.

Reply:
What server is this tested public ip 10.160.0.207? Better to use a domain name instead in case ip changed after new dhcp. Also when picking the server please use some stable server who acts like always up with same domain name.

Reply:
As test prepare for guest Virtual Networks, there was create ifcfg-eth1 under /etc/sysconfig/network/ on guest system, so after attach interface for guest, there will be created eth1 interface on guest, and then get ip address for eth1

Reply:
openqa.suse.de - 10.160.0.207, here, just make sure ping with ip address works . will updated code